### ISSUE TITLE
Using CrudAuth in order to save only own entities in create|update|delete

### ISSUE THREAD

**Issue by noneedinmagic** [nestjsx/crud#382]
Hello! I am testing new CrudAuth as a supposed way to get rid of my own implementation of service and controller which were extended before updating to 4.3.0+.

For getOneBase and getManyBase this CrudAuth decorator works perfectly.
However, I cannot find a way to restrict user of creating entities which he does not own.

For example: the user is an admin of organization which has departments.
He can select only employees who belong to departments of his organization (it is completed by eager joining of employee with department and with department's organization and then checking organization's ID).
Trying to select other organization's employee results in 404 which is perfectly okay.

However, this user still can create an employee with not own departmentId and get it's data on creation. However, he cannot select it later.

But this is a problem, cause some organization admins can fill my DB with garbage data on purpose or by accident.

For now I cannot find a way to verify saved entity by nested relation field. May anyone help me with solution of this?
Maybe I should implement some validation by creating custom class validator decorator which loads related entites from DB, check them and after that returns possible error?
Or a possiblity to add some specific query into transaction before saving new entity?

Thanks!

**Comment by zMotivat0r** [comment]
@noneedinmagic have you tried `persist`? 

>persist - a function that can return an object that will be added to the DTO on create, update, replace actions. Useful in case if you need to prevent changing some sensitive entity properties even if it's allowed in DTO validation.

```typescript
@CrudAuth({
   persist: (user) => ({ departmentId: user.departmentId })
})
```

**Comment by noneedinmagic** [comment]
> @noneedinmagic have you tried `persist`?
> 
> > persist - a function that can return an object that will be added to the DTO on create, update, replace actions. Useful in case if you need to prevent changing some sensitive entity properties even if it's allowed in DTO validation.
> 
> ```ts
> @CrudAuth({
>    persist: (user) => ({ departmentId: user.departmentId })
> })
> ```

Thanks for the reply! I've looked into it, but the docs too unclear for me.

> persist - a function that can return an object that will be added to the DTO on create, update, replace actions. Useful in case if you need to prevent changing some sensitive entity properties even if it's allowed in DTO validation.
There is no example of using it, so from description I've understood that it does something like `Object.assign(dto, persistObject)` just replacing values in sent data by hardcoded ones.

And my user does not have departmentId; user who is admin has method like `user.getOwnKeys('Organization')` which returns PKs of Organiaztion entities which are owned by this user.

So the user is owner of department if `department.organizationId == organization.id` and `user.getOwnKeys('Organization').includes(organization.id)`.

The case which you proposed would be working only if my req.user object had redundant linked data of all related institutions and then departments.

Also just checked, and yes, for deleteOne and updateOne decorator works, but for createOne, createMany and replaceOne I have no solution yet.

**Comment by zMotivat0r** [comment]
Yeah, Oleg, sorry for that. Please, keep in mind that I can't cover all possible scenarios, I think it's impossible.

**Comment by noneedinmagic** [comment]
> Yeah, Oleg, sorry for that. Please, keep in mind that I can't cover all possible scenarios, I think it's impossible.

Sure, that's why I am writing here with my scenario in attempt to find some solution :)

I would propose smth and even create PR if needed, but, unfortunately, I am somewhat new to NodeJs backend dev and this is my first serious project in JS on backend, most of my experience is from PHP, so it takes some time to adjust in this environment and learn all the things.

I thought about some custom validator for my class instance which I try to create, but then I need to populate related entities in order to check them before saving, or make a query to check their existence before saving my entity, and I don't understand how to do it without a lot of redundant injections of repositories and such stuff.

Injecting additional repos into my CRUD controller feels a bit redundant too.

I might implement smth like `entity.isOwnedBy(user: User): boolean`, but then for new instance of entity I still need a way to populate related models for current still-not-saved entity instance and then check their ownership (which also causes `isOwnedBy` method to be async which causes validation to be async too).

And there is another problem that for createMany this might cause a lot of redundant queries to check related models of created models. That's why in my own implementation of CrudService method createMany for now is blocked (until a solution would be found).

Maybe any ideas?

And yeah, it would be great if `persist` documentation would be more specific and/or with some examples.

**Comment by loicgeek** [comment]
@noneedinmagic 
@zMotivat0r 
Please check this #418
